plugins/@grouparoo/prometheus/src/actions/metrics.ts

```diff
@@ -0,0 +1,34 @@
+import { Action } from "actionhero";
+import { collectDefaultMetrics, Gauge, Registry } from "prom-client";
+import { Status } from "@grouparoo/core";
+
+//TODO: target specific metrics *OR* generalize to any metric
+const register = new Registry();
+const workersCount = new Gauge({
+  name: "workers_count",
+  help: "Number of workers in the cluster",
+  async collect() {
+    let status = await Status.getCurrent();
+    this.set(status.workers.cluster[0].metric.count);
+  },
+});
+register.registerMetric(workersCount);
+
+export default class PrometheusAction extends Action {
+  constructor() {
+    super();
+    this.name = "prometheus:getMetrics";
+    this.description = "Metrics endpoint supporting prometheus format";
+    this.middleware = ["authenticated-action"];
+    //@ts-ignore
+    this.permission = { topic: "system", mode: "read" };
+  }
```

Comment:
Try importing `import { AuthenticatedAction } from "@grouparoo/core/dist/classes/actions/authenticatedAction";` instead of `Action` from Actionhero - this will include the types of `this.permission` and handle the middleware for you

Reply:
Actually, it's probably better to also export `AuthenticatedAction` from `@grouparoo/core`'s index (like you did with Status) so it be used more simply.